src/EditorFeatures/CSharpTest/EditAndContinue/StatementMatchingTests.cs

```diff
@@ -863,6 +863,323 @@ public void LocalFunctionDefinitions()
+                { "{     int b()     {         int c()         {             int d() { return 0; };         };          return c();     };      return b(); }", "{     int b()     {         int c()         {             int d() { return H(0); };         };          return G(c());     };      return F(b()); }" },
+                { "int b()     {         int c()         {             int d() { return 0; };         };          return c();     }", "int b()     {         int c()         {             int d() { return H(0); };         };          return G(c());     }" },
+                { "{         int c()         {             int d() { return 0; };         };          return c();     }", "{         int c()         {             int d() { return H(0); };         };          return G(c());     }" },
+                { "int c()         {             int d() { return 0; };         }", "int c()         {             int d() { return H(0); };         }" },
+                { "{             int d() { return 0; };         }", "{             int d() { return H(0); };         }" },
+                { "int d() { return 0; }", "int d() { return H(0); }" },
+                { "{ return 0; }", "{ return H(0); }" },
+                { "return 0;", "return H(0);" },
+                { "return c();", "return G(c());" },
+                { "return b();", "return F(b());" }
+            };
+
+            expected.AssertEqual(actual);
+        }
+
+        [Fact]
+        public void LocalFunctions7()
+        {
+            var src1 = @"
+void G6(int a)
+{ 
+    int G5(int c) => /*1*/d;
+    F(G5);
+
+    void G4()
+    {
+        void G1(int x) => x;
+        int G3(int w)
+        { 
+            int G2(int c) => /*2*/d;
+            return G2(w);
+        }
+        F(G3);
+        F(G1);
+    };
+    F(G4);
+};
+";
+
+            var src2 = @"
+void G6(int a)
+{ 
+    int G5(int c) => /*1*/d + 1;
+    F(G5);
+
+    void G4()
+    {
+        int G3(int w)
+        { 
+            int G2(int c) => /*2*/d + 1;
+            return G2(w);
+        }
+        F(G3);
+        F(G1);
+        int G6(int p) => p *2;
+        F(G6);
+    };
+    F(G4);
+};
+";
+
+            var matches = GetMethodMatches(src1, src2);
+            var actual = ToMatchingPairs(matches);
+
+            var expected = new MatchingPairs
+            {
+                { "void G6(int a) {      int G5(int c) => /*1*/d;     F(G5);      void G4()     {         void G1(int x) => x;         int G3(int w)         {              int G2(int c) => /*2*/d;             return G2(w);         }         F(G3);         F(G1);     };     F(G4); }", "void G6(int a) {      int G5(int c) => /*1*/d + 1;     F(G5);      void G4()     {         int G3(int w)         {              int G2(int c) => /*2*/d + 1;             return G2(w);         }         F(G3);         F(G1);         int G6(int p) => p *2;         F(G6);     };     F(G4); }" },
```

Comment:
ok

---
In reply to: [132495769](https://github.com/dotnet/roslyn/pull/21396#discussion_r132495769) [](ancestors = 132495769)